test/gtest/ucp/test_ucp_tag.cc

```diff
@@ -423,22 +424,116 @@ class test_ucp_tag_limits : public test_ucp_tag {
         test_ucp_tag::init();
-        check_offload_support(m_test_offload);
+
+        if (m_test_offload) {
+            check_offload_support(m_test_offload);
+        }
     }
 
-    static void get_test_variants(std::vector<ucp_test_variant>& variants) {
-        add_variant_with_value(variants, get_ctx_params(), 0, "");
-        add_variant_with_value(variants, get_ctx_params(), 1, "offload");
+    static void get_test_variants(std::vector<ucp_test_variant>& variants) {\
+        ucp_params_t params = get_ctx_params();
+        params.features = UCP_FEATURE_TAG | UCP_FEATURE_AM;
+
+        add_variant_with_value(variants, params, 0, "");
+        add_variant_with_value(variants, params, 1, "offload");
     }
 
 protected:
     bool m_test_offload;
+
+    void check_rndv_startup_config(size_t exp_rndv_thresh,
+                                   size_t exp_rndv_intra_thresh,
+                                   size_t exp_rndv_inter_thresh)
+    {
+        ucp_context_config_t *cfg = &sender().worker()->context->config.ext;
+
+        EXPECT_EQ(exp_rndv_thresh, cfg->rndv_thresh);
+        EXPECT_EQ(exp_rndv_intra_thresh, cfg->rndv_intra_thresh);
+        EXPECT_EQ(exp_rndv_inter_thresh, cfg->rndv_inter_thresh);
+    }
+
+    void check_tag_rndv(size_t cfg_thresh)
+    {
+        ucp_ep_config_t *cfg = ucp_ep_config(sender().ep());
+
+        if (m_test_offload) {
+            /* There is a lower bound for rndv threshold with tag offload.
+             * We should not send messages smaller than this limit */
+            size_t min_rndv = ucp_ep_tag_offload_min_rndv_thresh(sender().ucph(),
+                                                                 &cfg->key);
+
+            /* If configured threshold is less than min_rndv, then expect exact
+             * min_rndv limit for short messages */
+            if (cfg_thresh < min_rndv) {
+                EXPECT_EQ(cfg->tag.offload.max_eager_short.memtype_on + 1,
+                          min_rndv);
+                EXPECT_EQ(cfg->tag.offload.max_eager_short.memtype_off + 1,
+                          min_rndv);
+            } else {
+                EXPECT_LE(cfg->tag.offload.max_eager_short.memtype_on + 1,
+                          cfg_thresh);
+                EXPECT_LE(cfg->tag.offload.max_eager_short.memtype_off + 1,
+                          cfg_thresh);
+            }
+        } else {
+            EXPECT_LE(cfg->tag.max_eager_short.memtype_on + 1, cfg_thresh);
+            EXPECT_LE(cfg->tag.max_eager_short.memtype_off + 1, cfg_thresh);
+        }
+    }
+
+    void check_am_rndv(size_t cfg_thresh)
+    {
+        ucp_ep_config_t *cfg = ucp_ep_config(sender().ep());
+
+        EXPECT_LE(cfg->am_u.max_eager_short.memtype_on + 1, cfg_thresh);
+        EXPECT_LE(cfg->am_u.max_eager_short.memtype_off + 1, cfg_thresh);
+        EXPECT_LE(cfg->am_u.max_reply_eager_short.memtype_on + 1, cfg_thresh);
+        EXPECT_LE(cfg->am_u.max_reply_eager_short.memtype_off + 1, cfg_thresh);
```

Comment:
maybe define a short wrapper function, like 
```
check_short_thresh (ucp_memtype_thresh_t *thresh, size_t cfg_thresh)
{
    EXPECT_LE(thresh->memtype_on + 1, cfg_thresh);
    EXPECT_LE(thresh->memtype_off + 1, cfg_thresh);
}
```
And use it for checks in all funcs of these test

Reply:
sure